### ISSUE TITLE
add heartbeat

### ISSUE THREAD

**Issue by serhtt** [Danielhiversen/PyXiaomiGateway#6]
https://github.com/Danielhiversen/PyXiaomiGateway/blob/c04e64b19e3b5ce48e43f7807a7b54586a03a663/PyXiaomiGateway/__init__.py#L146

elif cmd == 'report' or cmd == 'heartbeat':

**Comment by syssi** [comment]
The heartbeat was removed on purpose. Could you please explain why do you need/like the heartbeat?

**Comment by serhtt** [comment]
[https://cnbj2.fds.api.xiaomi.com/lumiaiot/common/gateway/%E7%BB%BF%E7%B1%B3%E7%BD%91%E5%85%B3%E5%B1%80%E5%9F%9F%E7%BD%91%E9%80%9A%E4%BF%A1%E5%8D%8F%E8%AE%AEV1.0.9_2017.08.01.doc](url)

**Comment by serhtt** [comment]
ERROR (Thread-13) [PyXiaomiGateway] Unknown multicast data : {'short_id': 54523, 'model': 'plug', 'cmd': 'heartbeat', 'sid': '158d0001dc60b3', 'data': '{"voltage":3600,"status":"off","inuse":"0","power_consumed":"146","load_power":"0.00"}'}

**Comment by syssi** [comment]
I've removed the error logging of heartbeat messages. We don't parse the heartbeat messages anymore because of incorrect content (for motion sensors). You can glean the full story here:

https://community.home-assistant.io/t/xiaomi-body-sensor-gateway-maybe-wrong-motion-detection/25130

Do you still need the heartbeat?

**Comment by serhtt** [comment]
I think this is just a problem for motion sensors. The others maybe requires this message.[https://forrestfung.gitbooks.io/lumi-gateway-local-api/content/](url)  This is the documentation

**Comment by syssi** [comment]
Can you provide an example? The documentation says the heartbeat is sent once per hour per device.

> {"Cmd": "heartbeat", "model": "magnet", "sid": "158d000065a271", "short_id": 25719, "data": "{\" status \ ": \" open \ "}" }.
Sub-device through the heartbeat tells the PC: sub-device normal work (heartbeat report frequency, the general sleep device is an hour once, the plug-in device is every 10 minutes). The sub-device heartbeat may contain the status of the child device. Note that this state is worth dealing with the specific use of the scene.
For example: when the window is closed
You can use this heartbeat (there may be a normal report status message is lost, heartbeat messages can be remedied).
For example: off the window open air conditioning scene
This can not use heartbeat. There may be people go, leaving the air conditioning off.
Heartbeat message and let air conditioning open will waste electricity.
In short, the use of heartbeat messages according to the scene requires the user to decide whether to use the heartbeat to do the trigger.

**Comment by syssi** [comment]
After meditation I think you are right. We should just skip the heartbeat of motion sensors. The 10 minutes heartbeat interval of a plug is data loss somehow. I will care about.

**Comment by syssi** [comment]
Do you accept this solution?

https://github.com/Danielhiversen/PyXiaomiGateway/pull/8/files
